secboot/encrypt_tpm_test.go

```diff
@@ -0,0 +1,99 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+// +build !nosecboot
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package secboot_test
+
+import (
+	"errors"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/secboot"
+)
+
+func (s *encryptSuite) TestFormatEncryptedDevice(c *C) {
+	for _, tc := range []struct {
+		initErr error
+		err     string
+	}{
+		{initErr: nil, err: ""},
+		{initErr: errors.New("some error"), err: "some error"},
+	} {
+		// create empty key to prevent blocking on lack of system entropy
+		myKey := secboot.EncryptionKey{}
+		for i := range myKey {
+			myKey[i] = byte(i)
+		}
+
+		calls := 0
+		restore := secboot.MockSbInitializeLUKS2Container(func(devicePath, label string, key []byte) error {
+			calls++
+			c.Assert(devicePath, Equals, "/dev/node")
+			c.Assert(label, Equals, "my label")
+			c.Assert(key, DeepEquals, myKey[:])
+			return tc.initErr
+		})
+		defer restore()
+
+		err := secboot.FormatEncryptedDevice(myKey, "my label", "/dev/node")
+		c.Assert(calls, Equals, 1)
+		if tc.err == "" {
+			c.Assert(err, IsNil)
+		} else {
+			c.Assert(err, ErrorMatches, tc.err)
+		}
+	}
+}
+
+func (s *encryptSuite) TestAddRecoveryKey(c *C) {
+	for _, tc := range []struct {
+		addErr error
+		err    string
+	}{
+		{addErr: nil, err: ""},
+		{addErr: errors.New("some error"), err: "some error"},
+	} {
+		// create empty key to prevent blocking on lack of system entropy
+		myKey := secboot.EncryptionKey{}
+		for i := range myKey {
+			myKey[i] = byte(i)
+		}
+
+		myRecoveryKey := secboot.RecoveryKey{15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0}
```

Comment:
Same suggestion as above to move the recovery key creation into the encryptionSuite setup

Reply:
I wonder if we should also create the keys in encrypSuite here (like we did in the gadget code). Somehting like:
```
diff --git a/secboot/encrypt_test.go b/secboot/encrypt_test.go
index 14860c4e78..4d149187be 100644
--- a/secboot/encrypt_test.go
+++ b/secboot/encrypt_test.go
@@ -28,12 +28,24 @@ import (
 	"github.com/snapcore/snapd/secboot"
 )
 
-type encryptSuite struct{}
+type encryptSuite struct {
+	mockedEncryptionKey secboot.EncryptionKey
+	mockedRecoveryKey   secboot.RecoveryKey
+}
 
 var _ = Suite(&encryptSuite{})
 
+func (s *encryptSuite) TestSetUp(c *C) {
+	// create empty key to prevent blocking on lack of system entropy
+	s.mockedEncryptionKey = secboot.EncryptionKey{}
+	for i := range s.mockedEncryptionKey {
+		s.mockedEncryptionKey[i] = byte(i)
+	}
+	s.mockedRecoveryKey = secboot.RecoveryKey{15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0}
+}
+
 func (s *encryptSuite) TestRecoveryKeySave(c *C) {
-	rkey := secboot.RecoveryKey{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 255}
+	rkey := s.mockedRecoveryKey
 	err := rkey.Save("test-key")
 	c.Assert(err, IsNil)
 	fileInfo, err := os.Stat("test-key")
diff --git a/secboot/encrypt_tpm_test.go b/secboot/encrypt_tpm_test.go
index 12b984c34d..a0b692b622 100644
--- a/secboot/encrypt_tpm_test.go
+++ b/secboot/encrypt_tpm_test.go
@@ -36,23 +36,17 @@ func (s *encryptSuite) TestFormatEncryptedDevice(c *C) {
 		{initErr: nil, err: ""},
 		{initErr: errors.New("some error"), err: "some error"},
 	} {
-		// create empty key to prevent blocking on lack of system entropy
-		myKey := secboot.EncryptionKey{}
-		for i := range myKey {
-			myKey[i] = byte(i)
-		}
-
 		calls := 0
 		restore := secboot.MockSbInitializeLUKS2Container(func(devicePath, label string, key []byte) error {
 			calls++
 			c.Assert(devicePath, Equals, "/dev/node")
 			c.Assert(label, Equals, "my label")
-			c.Assert(key, DeepEquals, myKey[:])
+			c.Assert(key, DeepEquals, s.mockedEncryptionKey[:])
 			return tc.initErr
 		})
 		defer restore()
 
-		err := secboot.FormatEncryptedDevice(myKey, "my label", "/dev/node")
+		err := secboot.FormatEncryptedDevice(s.mockedEncryptionKey, "my label", "/dev/node")
 		c.Assert(calls, Equals, 1)
 		if tc.err == "" {
 			c.Assert(err, IsNil)
@@ -70,25 +64,17 @@ func (s *encryptSuite) TestAddRecoveryKey(c *C) {
 		{addErr: nil, err: ""},
 		{addErr: errors.New("some error"), err: "some error"},
 	} {
-		// create empty key to prevent blocking on lack of system entropy
-		myKey := secboot.EncryptionKey{}
-		for i := range myKey {
-			myKey[i] = byte(i)
-		}
-
-		myRecoveryKey := secboot.RecoveryKey{15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0}
-
 		calls := 0
 		restore := secboot.MockSbAddRecoveryKeyToLUKS2Container(func(devicePath string, key []byte, recoveryKey [16]byte) error {
 			calls++
 			c.Assert(devicePath, Equals, "/dev/node")
-			c.Assert(recoveryKey[:], DeepEquals, myRecoveryKey[:])
-			c.Assert(key, DeepEquals, myKey[:])
+			c.Assert(recoveryKey[:], DeepEquals, s.mockedRecoveryKey[:])
+			c.Assert(key, DeepEquals, s.mockedEncryptionKey[:])
 			return tc.addErr
 		})
 		defer restore()
 
-		err := secboot.AddRecoveryKey(myKey, myRecoveryKey, "/dev/node")
+		err := secboot.AddRecoveryKey(s.mockedEncryptionKey, s.mockedRecoveryKey, "/dev/node")
 		c.Assert(calls, Equals, 1)
 		if tc.err == "" {
 			c.Assert(err, IsNil)
```
but it seems less important here as we only use myKey twice and the recovery key once